internal/v1/handler.go

```diff
@@ -513,6 +513,32 @@ func convertIgnoreImageTypeToSlice(ignoreImageTypes *[]ImageTypes) []string {
 	return stringSlice
 }
 
+func (h *Handlers) fetchComposes(orgId string, blueprintId *uuid.UUID, blueprintVersion *int, since time.Duration, limit, offset int, ignoreImageTypeStrings []string) ([]db.ComposeEntry, int, error) {
+	if blueprintId != nil {
+		return h.server.db.GetBlueprintComposes(orgId, *blueprintId, since, limit, offset, ignoreImageTypeStrings)
+	}
+	return h.server.db.GetComposes(orgId, since, limit, offset, ignoreImageTypeStrings)
+}
+
+func (h *Handlers) formatComposeLinks(specVersion string, count, limit, offset int, blueprintId *uuid.UUID, blueprintVersion *int) (string, string) {
+	lastOffset := count - 1
+	if lastOffset < 0 {
+		lastOffset = 0
+	}
+	path := fmt.Sprintf("%v/v%v/composes", RoutePrefix(), specVersion)
+	if blueprintId != nil {
+		path = fmt.Sprintf("%v?blueprint_id=%s&", path, blueprintId.String())
+		if blueprintVersion != nil {
+			path = fmt.Sprintf("%vblueprint_version=%d&", path, *blueprintVersion)
+		}
+	} else {
+		path = path + "?"
+	}
+	first := fmt.Sprintf("%voffset=0&limit=%v", path, limit)
+	last := fmt.Sprintf("%voffset=%v&limit=%v", path, lastOffset, limit)
+	return first, last
+}
```

Comment:
Is it possible that I don’t see a unit test for this function? :-D

Reply:
Nooo ! Busted ! :fearful: 
I can add one I guess :+1:

Reply:
I've added it in the handler test, but I've got rid of this fucntion now